### ISSUE TITLE
[BUG] shaman Searing Totem targeting

### ISSUE THREAD

**Issue by nochevnoy** [ClassicWoWCommunity/cata-classic-bugs#1462]
#### 1.1. Description
A totem that selects a target without any logic and does not change the target when Stormstrike debuff is applied to it.
  #### 1.2. How to Reproduce
           1. taunt 1st mob
           2. place searing totem
           3. totem starts hitting 1st mob
           4. pull another mob
           5. apply stormstrike to it
           6. totem won't switch target
  #### 1.3  Source Material

#### 2.1. Description
totem should switch target
  #### 2.2 Source Material
https://www.wowhead.com/cata/spell=3599/searing-totem
<img width="324" alt="изображение" src="https://github.com/user-attachments/assets/e443ccb0-7fd3-4f02-8274-0c069ea95bc3">

**Comment by freddaxyz** [comment]
i gotta say, the more testing you do on this totem the more strange things you discover. 

but to keep it relatively concise, as far as i can tell, it does prioritize stormstrike over targets without debuffs what so ever. however, it does not prioritize its attacks to a stormstriked target (or stormstriked + flame shocked) over a flame shocked one, only over one without either debuff, which is what i'm guessing you're experiencing. so this means that any time you spread your flame shock, as the spec is intended to be played, you're lucky if your searing totem is hitting the right target. what is the priority really doing for me if every target is flame shocked and it's just picked at random between those targets? (it isn't actually random as i've discovered, but it effectively is for the player as it's not something we can change)

i can't say for certain if this is a bug, but i don't see any negatives with changing this because it fixes almost all the searing totem issues for enhancement and it's not relevant to other specs

**Comment by Artemishowl** [comment]
I agree on this being "SomeChanges" and will require more digging before validity, making a ticket

**Comment by nochevnoy** [comment]
any chance to be fixed till cata ends?
